packages/table/src/table.tsx

```diff
@@ -1995,12 +1998,24 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
             // doing, we add the size of the header elements, which are not technically part of the
             // "grid" concept (the grid only consists of body cells at present).
             if (didScrollTopChange) {
-                const topCorrection = this.shouldDisableVerticalScroll() ? 0 : this.columnHeaderElement.clientHeight;
-                this.scrollContainerElement.scrollTop = nextScrollTop + topCorrection;
+                const topCorrection =
+                    this.shouldDisableVerticalScroll() || this.columnHeaderElement == null
```

Comment:
Would prefer not to add the `this.columnHeaderElement == null` check until we move forward on #1606. Then we'll do it all at once.